lib/ansible/modules/network/nuage/nuage_vspk.py

```diff
@@ -0,0 +1,1089 @@
+        if set_output:
+            self.result['id'] = entity.id
+            self.result['entities'].append(entity.to_dict())
+
+    def _create_entity(self, entity_class, parent, properties):
+        """
+        Creates a new entity in the parent, with all properties configured as in the file
+        :param entity_class: The class of the entity
+        :param parent: The parent of the entity
+        :param properties: The set of properties of the entity
+        :return: The entity
+        """
+        entity = entity_class(**properties)
+        try:
+            parent.create_child(entity)
+        except BambouHTTPError as error:
+            self.module.fail_json(msg='Unable to create entity: {0}'.format(error))
+        self.result['changed'] = True
+        return entity
+
+    def _save_entity(self, entity):
+        """
+        Updates an existing entity
+        :param entity: The entity to save
+        :return: The updated entity
+        """
+        try:
+            entity.save()
+        except BambouHTTPError as error:
+            self.module.fail_json(msg='Unable to update entity: {0}'.format(error))
+        self.result['changed'] = True
+        return entity
+
+    def _delete_entity(self, entity):
+        """
+        Deletes an entity
+        :param entity: The entity to delete
+        """
+        try:
+            entity.delete()
+        except BambouHTTPError as error:
+            self.module.fail_json(msg='Unable to delete entity: {0}'.format(error))
+        self.result['changed'] = True
+
+    def _wait_for_job(self, entity):
+        """
+        Waits for a job to finish
+        :param entity: The job to wait for
+        """
+        running = False
+        if entity.status == 'RUNNING':
+            self.result['changed'] = True
+            running = True
+
+        while running:
+            time.sleep(1)
+            entity.fetch()
+
+            if entity.status != 'RUNNING':
+                running = False
+
+        self.result['entities'].append(entity.to_dict())
+        if entity.status == 'ERROR':
+            self.module.fail_json(msg='Job ended in an error')
+
+
+def main():
+    """
+    Main method
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            auth=dict(required=True, type='dict', no_log=True),
+            type=dict(required=True, type='str'),
+            id=dict(default=None, required=False, type='str'),
+            parent_id=dict(default=None, required=False, type='str'),
+            parent_type=dict(default=None, required=False, type='str'),
+            state=dict(default=None, choices=['present', 'absent'], type='str'),
+            command=dict(default=None, choices=SUPPORTED_COMMANDS, type='str'),
+            match_filter=dict(default=None, required=False, type='str'),
+            properties=dict(default=None, required=False, type='dict'),
+            children=dict(default=None, required=False, type='list')
+        ),
```

Comment:
https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/developing_modules_general.rst#main-and-ansiblemodule-argument-spec